files/stress_tests.sh

```diff
@@ -0,0 +1,281 @@
+    message "Unmounting ${ssd_tests_folder} after trap triggered..."
+    umount "${ssd_tests_folder}"
+  fi
+  if [ -n "${ssd_volume_group}" ]; then
+    if [ -e "/dev/mapper/${ssd_volume_group}-stress" ]; then
+      message "Removing Logical Volume '/dev/mapper/${ssd_volume_group}-stress' after trap triggered..."
+      lvremove "/dev/mapper/${ssd_volume_group}-stress" --force
+    fi
+  fi
+}
+
+function usage() {
+  echo "${APP_NAME} \
+    [--help|-h|-?] \
+    [--stress-ng-duration|-t 'Streess-NG test duration'] \
+    [--fio-test-duration|-e 'FIO test duration'] \
+    [--hdd-bytes|-d 'Bytes per HDD worker'] \
+    [--vm-bytes|-b 'Bytes per VM worker'] \
+    [--cpu|-c 'CPU workers'] \
+    [--vm|-v 'VM workers'] \
+    [--hdd-path|-w 'HDD workers'] \
+    [--no-ssd-test|-n] \
+    [--ssd-block-size|-k 'Size of the blocks for SSD tests' \
+    [--ssd-workers|-o 'Number of workers in SSD tests'
+    [--ssd-tests-folder|-f 'Folder to mount and place temporary files.'
+    [--ssd-volume-group|-g 'Logical Group where the SSD tests are running. Check that it contained in SSD disks.' \
+    [--log-file|-l 'Logs filename']"
+}
+
+function check_ssd() {
+  # Check if all the disks are SSD (not rotational)
+  result=0
+  while read -r DISK
+  do
+    test -e "${DISK}/queue/rotational" && result=$((result + $(cat "${DISK}/queue/rotational")))
+  done <<< "$(find /sys/block/ -maxdepth 1)"
+  return "${result}"
+}
+
+function need_fio_test() {
+  # If there is no volume group called vg_instances don't run fio test
+  if /usr/sbin/vgdisplay vg_instances; then
+    message "There is a volume group called vg_instances"
+  else
+    message "There is NOT a volume group called vg_instances. Not running FIO test"
+    return 1
+  fi
+
+  # Check if there are SSD disks behind a RAID controller
+  if [ -e /opt/MegaRAID/MegaCli/MegaCli64 ]; then
+    message "MegaRAID CLI is installed"
+    if [ "$(/opt/MegaRAID/MegaCli/MegaCli64 -PDList -aALL | grep -i solid)" != "" ]; then
+      message "MegaRAID CLI reports SSD disks"
+      return 0
+    fi
+  fi
+
+  # Check if disks are SSD (not rotational) according to the kernel
+  if check_ssd; then
+    message "Kernel reports SSD disks"
+    return 0
+  fi
+
+  return 1
+}
+function message() {
+  text="${1}"
+  if [ "${text}" != "" ]; then
+    current_date="$(date +'%Y-%m-%d.%H.%M.%S%z')"
+    if [ -n "${log_file}" ]; then
+      echo "${current_date} ${text}" >> "${log_file}"
+    fi
+    echo "${current_date} ${text}"
+    logger -t "${APP_NAME}" "${text}"
+  fi
+}
+
+function run_and_log() {
+  temp_log_file=$(mktemp /tmp/tmp.log.XXXXX)
+  message "Running: $*"
+  # shellcheck disable=SC2048,SC2086
+  eval $* >> "${temp_log_file}" 2>&1
+  return_code="$?"
+  message "Returned code: ${return_code}"
+  while read -r log_line
+  do
+    message "${log_line}"
+  done <<< "$(cat "${temp_log_file}")"
+  return "${return_code}"
+}
+
+log_file="${HOME}/${APP_NAME}_$(date +'%Y-%m-%d.%H.%M.%S%z').log"
```

Comment:
Adding the date into log file and without removing it can create affect the HOME folder. An option to save the log files into /tmp.

Reply:
Changed